versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraBackend.java

```diff
@@ -329,16 +334,40 @@ boolean executeCas(BoundStatement stmt) {
   }
 
   ResultSet execute(BoundStatement stmt) {
-    return session.execute(stmt);
+    try {
+      return session.execute(stmt);
+    } catch (DriverException e) {
+      throw unhandledException(e);
+    }
   }
 
   CompletionStage<AsyncResultSet> executeAsync(BoundStatement stmt) {
     return session.executeAsync(stmt);
   }
 
-  void handleDriverException(DriverException e) {
-    if (e instanceof CASWriteUnknownException) {
-      logCASWriteUnknown((CASWriteUnknownException) e);
+  static RuntimeException unhandledException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      return new UnknownOperationResultException(e);
+    } else if (e instanceof DriverTimeoutException
+        || e instanceof NodeUnavailableException
+        || e instanceof AllNodesFailedException) {
+      return new UnknownOperationResultException(e);
+    } else {
+      return e;
+    }
+  }
+
+  static void handleDriverException(DriverException e) {
+    if (e instanceof QueryExecutionException) {
+      if (e instanceof CASWriteUnknownException) {
+        logCASWriteUnknown((CASWriteUnknownException) e);
```

Comment:
Why don't we throw in this case?

Reply:
Yea - with this change we can remove this special handling.